File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/javascript/fibonacci.js
const fibonacci = (num) => parseInt((((((1+(5**0.5))/2)**num)-(((1-(5**0.5))/2)**num))/(5**0.5)))

// function fibonacci(num){
//   let sum = 0
//   let x = 0
//   let y = 1
//   if (num < 2) return num;

//   for(let i = 0; i < num; i++){
//     sum = x + y
//     y = x
//     x = sum
//   }
//   return sum
// }
if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file 

// def fibonacci
// add argument to binet's formula
// convert results to integer
// done

// And a written explanation of your solution

// I first researched fibonacci as I hadn't used it since highschool, 
// after researching I researced a formula that calcutates the exact
// number of each fibonacci sequence number based on position, I found
// Binet's formula for calculating finbonacci sequence numbers. I then 
// plugged that in and tested it to see results.

File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  (((((1+(5**0.5))/2)**num)-(((1-(5**0.5))/2)**num))/(5**0.5)).to_i 
end

# def fib(num)
#   x=1
#   y=0
#   sum = 0
#   return num if num < 2
#   (num-1).times do 
#     sum = x + y
#     y = x
#     x = sum
#   end
#   sum
# end

# def fibonacci(number)
#   if number < 2
#     number
#   else
#     fibonacci(number - 1) + fibonacci(number - 2)
#   end
# end


if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)
  

  # Don't forget to add your own!

  puts "Expecting: 1597"
  puts "=>", fibonacci(17)

  puts

  puts "Expecting: 46368"
  puts "=>", fibonacci(24)

  puts
end

# Please add your pseudocode to this file 

# def fibonacci
#   add argument to binet's formula
#   convert results to integer
# done

# And a written explanation of your solution

# I first researched fibonacci as I hadn't used it since highschool, 
# after researching I researced a formula that calcutates the exact
# number of each fibonacci sequence number based on position, I found
# Binet's formula for calculating finbonacci sequence numbers. I then 
# plugged that in and tested it to see results.
